ident/identifier.go

```diff
@@ -43,68 +41,32 @@ func StringID(str string) ID {
 
 type id struct {
 	data checked.Bytes
-	hash Hash
```

Comment:
Any concern about removing this, or is fnv64 fast enough that it doesn't matter if we compute hashes multiple times?

Reply:
Fair enough, that seems reasonable to me. 

As an aside: why fnv over xx? All the stuff I can find online suggests xxhash is all around better for our kinda workload (e.g. [1](https://aras-p.info/blog/2016/08/02/Hash-Functions-all-the-way-down/) [2](https://github.com/Cyan4973/xxHash/)).

Reply:
Hm, yeah xxhash does seem to be quite fast:
```go
$ go test -v -bench . -run none
goos: darwin
goarch: amd64
pkg: github.com/m3db/m3x/hash/fnv
BenchmarkFnv-4          30000000                54.3 ns/op
BenchmarkMd5-4          10000000               165 ns/op
BenchmarkXxhash-4       100000000               20.3 ns/op
PASS
ok      github.com/m3db/m3x/hash/fnv    5.583s
```

Also seems to be used in Influx and Prometheus:
https://github.com/cespare/xxhash#projects-using-this-package

I'll pull it in, MIT licensed so should be good.